### PR TITLE
[new release] lambdapi (2.3.1)

### PR DESCRIPTION
**Pull request by fblanqui**
Proof assistant for the λΠ-calculus modulo rewriting

- Project page: <a href="https://github.com/Deducteam/lambdapi">https://github.com/Deducteam/lambdapi</a>

### Fixed

- Opaque definitions are not kept in memory and in lpo files anymore
- A few bug fixes.

### Changed

- Why3 dependency updated to 1.6.

### CODE REVIEW

**Comment by mseri** [comment]
This version is failing on macos and ppc64: https://opam-ci.ci3.ocamllabs.io/github/ocaml/opam-repository/commit/9271226fcba9aa04723ffb2534424501d1eb5405

Do you know what is going on?

**Comment by fblanqui** [comment]
@gabrielhdt @TheoWinterhalter do you have any idea if you use a Mac?

**Comment by TheoWinterhalter** [comment]
Do I understand correctly that it's complaining that a file may not be found? If so it might a problem related to case sensitivity? Like maybe you renamed `foo.dk` to `Foo.dk`? This kind of stuff can be a cause for a lot of problems on macOS.

**Comment by TheoWinterhalter** [comment]
Ah yes, in tests you have both `bool.lp` and `Bool.lp`. Same for `nat.lp` and `Nat.lp`. I suspect this might be the cause.

**Comment by fblanqui** [comment]
Ah, thank you. I didn't know about this. But it didn't show up in previous versions... @mseri Something has been changed on opam CI ?

**Comment by fblanqui** [comment]
Hi. I modified the package by changing the names of some test files. I solved all issues but one. I don't understand the last error. It doesn't do test ` 78   OK/rule_order.lp`. I don't understand why, and there is no error message.
https://opam-ci.ci3.ocamllabs.io/github/ocaml/opam-repository/commit/679a2d941fb40f78f20fc911079e72f35a02fa7c/variant/extras,ppc64-ocaml-4.14,lambdapi.2.3.1,tests

**Comment by fblanqui** [comment]
@mseri I don't know what else to do. Is it mandatory to pass this test to merge this PR?

**Comment by TheoWinterhalter** [comment]
There is no error message, but there is something that looks like a prefix of one:
```
# File "tests/dune", line 2, characters 8-13:
# 2 |  (names ok_ko rewriting purity kernel)
#             ^^^^^
```
I don't know what this means however. Also this time it seems to only affect PowerPC, so I have no idea about it myself.

**Comment by fblanqui** [comment]
Hi @mseri . I have no way to know what is the problem and fix it. Is it mandatory to pass the test on PowerPC to merge this PR? Please let me know.

**Comment by mseri** [comment]
Not a problem. Do you prefer for me to disable the tests or make the architecture unavailable? The kind of failure makes me think that it may actually segfault there, so I am more prone to the latter.

**Comment by fblanqui** [comment]
Ok let's do that (make arch unavailable). Thank you!

File: packages/lambdapi/lambdapi.2.3.1/opam [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+    "-p"
+    name
+    "-j"
+    jobs
+    "@install"
+    "@runtest" {with-test}
+    "@doc" {with-doc}
+  ]
+]
+url {
```
Comment (mseri) [inline packages/lambdapi/lambdapi.2.3.1/opam:79]:
```suggestion
available: arch != "ppc64"
url {
```

**Comment by fblanqui** [comment]
@mseri Do you know what I should change in my dune-project file if I want my opam file to be generated by dune automatically?

**Comment by kit-ty-kate** [comment]
> Do you know what I should change in my dune-project file if I want my opam file to be generated by dune automatically?

I don't think this is supported by dune. I would discourage using dune autogeneration but if you really want to keep using it you can create a `lambdapi.opam.template` and put just the `available` field there. See https://dune.readthedocs.io/en/stable/opam.html#opam-template

Thanks

**Pull request merged**